gensim/test/test_sklearn_api.py

```diff
@@ -1170,7 +1170,8 @@ def setUp(self):
         ]
 
     def testCompareToOld(self):
-        old_phrases_transformer = pickle.load(datapath("phrases_transformer.pkl"))
+        with open(datapath("phrases_transformer.pkl"), "rb") as old_phrases_transformer_pkl:
+            old_phrases_transformer = pickle.load(old_phrases_transformer_pkl)
         doc = phrases_sentences[-1]
         phrase_tokens = old_phrases_transformer.transform(doc)[0]
```

Comment:
The old `PhrasesTransformer` didn't have the `self.phraser` attribute which is checked in the new `transform()` method. This causes this test to fail. It seems that the new `transform()` method is being called here instead of the old one. Do I need to take extra care of object methods being pickled correctly?

Reply:
" old one" no more exists, you have an old model (like a key-value store of data) and new code. 
> Do I need to take extra care of object methods being pickled correctly?

Of course, see https://docs.python.org/2/library/pickle.html#pickling-and-unpickling-normal-class-instances